src/Numeric/Sundials.hs

```diff
@@ -266,11 +284,16 @@ withCConsts ODEOpts{..} OdeProblem{..} = runContT $ do
           funIO :: OdeRhsCType
           funIO t y f _ptr = do
             sv <- peek y
-            r <- fun t (sunVecVals sv)
-            poke f $ SunVector { sunVecN = sunVecN sv
-                               , sunVecVals = r
-                               }
-            return 0
+
+            -- Save the exception (if any)
+            saveExceptionContext exceptionRef $ do
+              r <- fun t (sunVecVals sv)
+
+              -- Note: the following operation will force "r"
```

Comment:
will it? You removed `resM <- try (evaluate (Control.DeepSeq.force $ coerce f t y))` from `funIO`

Reply:
Yes, indeed. But actually, the `poke` thing is converting the complete data structure (i.e. the `SunVector`) to a C representation, so it need to force its content. Hence no need for `evaluate/force` which will do double work.